.github/workflows/windows.yml

```diff
@@ -135,9 +135,10 @@ jobs:
           # Should save cache only if run in the master branch of the base repo
           # github.ref_name is 'ref/PR_#' in case of the PR, and 'branch_name' when executed on push
           save: ${{ github.ref_name == 'master' && 'true' || 'false'  }}
-          key: ${{ github.job }}-windows
+          append-timestamp: true
+          key: ${{ github.job }}-${{ runner.os }}
           restore-keys: |
-            ${{ github.job }}-windows
+            ${{ github.job }}-${{ runner.os }}
```

Comment:
I suggest giving it an even more specific name (like `${{ github.job }}-${{ runner.os }}-native/full-valiation`) so that it would not get caches from `${{ github.job }}-${{ runner.os }}-cc` and alike as the key matching is based on the substring.

Reply:
fixed